### PR TITLE
Remove gotos in vehicle.cpp

### PR DESCRIPTION
**Pull request by frutiemax**

### CODE REVIEW

**Comment by frutiemax** [comment]
Do not merge, there is an infinite loop in one of the commits...

**Comment by janisozaur** [comment]
This issue is filed as https://github.com/OpenRCT2/OpenRCT2/issues/11394 and there @glaubertadeu made the changes already, long time ago, but never submitted the pr. The code is available in their fork.

I'd like to test this pr, but it will take a bit of time until I'm ready, maybe even few weeks.

**Comment by frutiemax** [comment]
git.exe bisect bad
80bc70960b949253895fe3fb693ebd3ff1c0141a is the first bad commit
commit 80bc70960b949253895fe3fb693ebd3ff1c0141a
Author: frutiemax <lucas.malo.belanger@gmail.com>
Date:   Fri Dec 23 21:52:52 2022 -0500

Remove goto label loc_6DC985

src/openrct2/ride/Vehicle.cpp | 114 +++++++++++++++++++++++++++++++++---------
1 file changed, 91 insertions(+), 23 deletions(-)

**Comment by tupaschoal** [comment]
Note even though you force pushed yesterday, it has some merge conflicts with Vehicle

**Comment by Gymnasiast** [comment]
I checked the commits up to and including "Remove goto label loc_6DC743". The next one is quite a bit more complicated. I might split the PR in two.

**Comment by duncanspumpkin** [comment]
@frutiemax could you rebase this and split it into two PRs. Just the first 8 or so commits in this one.

**Comment by Gymnasiast** [comment]
I would recommend putting everything up to and including `loc_6DC743` into the first one - I have already checked these and found them to be correct.

**Comment by tupaschoal** [comment]
Ping @frutiemax

**Comment by Gymnasiast** [comment]
Please keep this open, I can take take some things over if needed.

File: src/openrct2/ride/Vehicle.cpp
Comment (Broxzier) [inline src/openrct2/ride/Vehicle.cpp:8118]: Accidental blank line

File: src/openrct2/ride/Vehicle.cpp
Comment (Broxzier) [inline src/openrct2/ride/Vehicle.cpp:8359]: Same as previous comment.

**Review by Broxzier — CHANGES_REQUESTED** [review]
Code looks mostly good and the replays pass which is promising, but I'd like to verify this with a couple of parks with lots of rides to make ensure that the logic is identical. Recording a 3-minute replay of a park with lots of different rides should cover most of this.

File: src/openrct2/ride/Vehicle.cpp
Comment (Broxzier) [inline src/openrct2/ride/Vehicle.cpp:8693]: Same as previous comment.

**Comment by Broxzier** [comment]
@frutiemax Will you be able to tackle the comments above and get this in a mergable state?

**Comment by Gymnasiast** [comment]
I might take a stab, please don’t close this just yet.

**Comment by Gymnasiast** [comment]
Keep this open please.

**Comment by Gymnasiast** [comment]
Keep this open please.

**Comment by Gymnasiast** [comment]
Keep this open please.

**Comment by Gymnasiast** [comment]
I decided to split this PR into two. The first 10 commits (all of which I have been able to check) are here: https://github.com/OpenRCT2/OpenRCT2/pull/21111

After that one is merged, I intend to do the other five.

**Comment by tupaschoal** [comment]
Superseded by #21111 and #22712, likely too old to reuse anyway, thus closing this.

**Pull request closed without merging**
